app/controllers/application_controller.rb

```diff
@@ -1,14 +1,34 @@
 class ApplicationController < ActionController::Base
+  
   protect_from_forgery with: :exception
   include SessionsHelper
-  
   helper_method :current_order
 
   def current_order
-    if !session[:order_id].nil?
-      Order.find(session[:order_id])
+    if !session[:order_id].nil? && !search_order.nil?
+      order = Order.find(session[:order_id])
```

Comment:
not sure I get what's happening in here, could you try to explain?

Reply:
Also, `session[:order_id].present? && search_order.present?` is nicer

Reply:
It is searching if the user didn't place the order.If he did, it creates a new order for the user, else the products will be placed on the current order.